app/controllers/concerns/posts_controller.rb

```diff
@@ -0,0 +1,57 @@
+class PostsController < ApplicationController
+  def index
+    @posts = Post.includes(:author).where(author_id: params[:user_id])
+    @user = User.find(params[:user_id])
+    @posts = @posts.paginate(page: params[:page], per_page: 3)
```

Comment:
- [ ] To fix the N+1 problem when fetching all posts and their comments for a user in the `posts#index` actions you should use `@posts = @user.posts.includes(:comments)` instead of what you did here `@posts = Post.includes(:author).where(author_id: params[:user_id])`